### PR TITLE
HPCC-20036 Various sasha cli fixes.

### PR DESCRIPTION
**Pull request by jakesmith**
1) action=list with no other options crashes.
Cause: tree not loaded, only the id is wanted.
2) SCA_GET doesn't return a count.
Cause: addId not called.
3) output format variables missing, e.g. wuid, command, state.
Missed during refactoring?

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by jakesmith** [comment]
NB: this is based of the unmerged commit for HPCC-20034, which is needed to allow sasha CLI to run.

@wangkx - please review.

File: dali/sasha/saarch.cpp [hunk 1]
```diff
@@ -293,19 +299,22 @@ void WUiterate(ISashaCommand *cmd, const char *mask)
...
-                else
-                    found = false;
-                if (found)
+                    val = wu->queryProp("@state");
+                else if (strieq(outputField,"state"))
+                    val = wu->queryProp(dfu ? "Progress/@state" : "@state");
+                else if (strieq(outputField, "command"))
+                    val = wu->queryProp("@command");
+                else if (strieq(outputField, "wuid"))
+                    val = wu->queryName();
```
Comment (jakesmith) [inline dali/sasha/saarch.cpp]: @wangkx - I could see why the above output format options had been removed during the HPCC-11822 refactoring..
do you know why?

Comment (wangkx) [reply dali/sasha/saarch.cpp]: In the HPCC-11822, I assume that we always want to output wuid (to know which wu information belongs to). The assumption could be wrong.

Comment (jakesmith) [reply dali/sasha/saarch.cpp]: The wuid is a prefix, but "wuid" was an option in the output fields, so removing might break backward compatibility.

**Review by wangkx — COMMENTED** [review]
@jakesmith one question

File: dali/sasha/saarch.cpp [hunk 2]
```diff
@@ -293,19 +299,22 @@ void WUiterate(ISashaCommand *cmd, const char *mask)
...
                 else if (strieq(outputField, "jobname"))
-                    wu->getProp("@jobName",val);
+                    val = wu->queryProp("@jobName");
                 else if (strieq(outputField,"state"))
-                    wu->getProp("@state", val);
-                else
-                    found = false;
-                if (found)
+                    val = wu->queryProp("@state");
+                else if (strieq(outputField,"state"))
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply dali/sasha/saarch.cpp]: seems so, I'll remove lines 309+310

**Comment by jakesmith** [comment]
@richardkchapman - ready to merge.

**Pull request merged**